Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -2493,4 +2510,20 @@ private void parseAdminInstall(CommandSender sender) throws NoPermissionExceptio
 			TownyMessaging.sendMsg(sender, Translatable.of("msg_setup_success"));
 		});
 	}
+
+	private void parseAdminTownForSaleCommand(CommandSender sender, Town town, String[] arg) throws TownyException {
+		if (arg.length == 0)
+			throw new TownyException(Translatable.of("msg_error_must_be_num"));
+
+		double forSalePrice = Double.parseDouble(arg[0]);
+		TownCommand.setTownForSale(town, forSalePrice, true);
+
+		TownyMessaging.sendMsg(sender, Translatable.of("msg_town_forsale", town.getName(), TownyEconomyHandler.getFormattedBalance(forSalePrice)));
+	}
+
+	private void parseAdminTownNotForSaleCommand(CommandSender sender, Town town) {
+		TownCommand.setTownNotForSale(town, true);
+
+		TownyMessaging.sendMsg(sender, Translatable.of("msg_town_notforsale", town.getName()));

```

Comment:
It might be a good idea to also send a town message, so the town members know they are not for sale anymore.